actix-files/src/named.rs

```diff
@@ -95,7 +95,13 @@ impl NamedFile {
             };
             let cd = ContentDisposition {
                 disposition: disposition_type,
-                parameters: vec![DispositionParam::Filename(filename.into_owned())],
+                parameters: vec![
+                    DispositionParam::FilenameExt(ExtendedValue {
```

Comment:
Still one problem: by replacing Filename(`filename`) with FilenameExt(`filename*`), it brings risks that some old user agent implementations cannot understand `filename*`.

RFC6266 suggests that:
> When a "filename*" parameter is sent, to also generate a "filename" parameter as a fallback for user agents that do not support the "filename*" form, if possible.  This can be done by substituting characters with US-ASCII sequences (e.g., Unicode character point U+00E4 (LATIN SMALL LETTER A WITH DIARESIS) by "ae").  Note that this may not be possible in some locales.

This is not implemented in this PR, because I am not sure how to "substituting characters" generally. Do you have any  ideas?

Reply:
Or just leave `filename` parameters intact? Say,
```rust
let cd = ContentDisposition {
    disposition: disposition_type,
    parameters: vec![
        DispositionParam::Filename(filename.into_owned()),
        DispositionParam::FilenameExt(ExtendedValue {
            charset: Charset::Ext(String::from("UTF-8")),
            language_tag: None,
            value: filename.as_bytes().to_vec(),
        }),
    ],
};
```
so that there are both `filename*` and `filename` which might contains invalid charaters.

Reply:
> This is not implemented in this PR, because I am not sure how to "substituting characters" generally. Do you have any ideas?

It seems that you have to replace/translate Unicode characters to equivalent in ASCII that and even so they say not every Unicode char has an equivalent in ASCII.
So I prefer your second proposal.

Reply:
> It seems that you have to replace/translate Unicode characters to equivalent in ASCII that and even so they say not every Unicode char has an equivalent in ASCII.

You are right. As suggested by @gumblex, iconv has a great [TRANSLIT](https://spin.atomicobject.com/2011/07/13/some-useful-iconv-functionality/) function, which could "convert characters in the origin encoding to the closest possible matching character in the target encoding". But the problem is still "not every Unicode char has an equivalent in ASCII" and in which case those chars are barely replaced by Unicode Replacement Character. So I also prefer the second way.

Reply:
Update:
As suggested by @lambdaq, there is something like https://github.com/chowdhurya/rust-unidecode which provides decent transliteration from Unicode to ASCII. But its functionality is still limited. And I do not think it is worth to introduce a new dependency. Just write it down here for future reference.

Reply:
Hi, I am wondering what if I suggested anything at all. I dont write Rust for now but I do want to learn this language.